fineract-provider/src/test/java/org/apache/fineract/infrastructure/core/LiquibaseStepDefinitions.java

```diff
@@ -0,0 +1,131 @@
+
+            given(databaseStateVerifier.isLiquibaseDisabled()).willReturn(this.disabled);
+            given(fineractProperties.getTenant()).willReturn(new FineractProperties.FineractTenantProperties());
+            given(liquibaseFactory.create(defaultTenantDataSource, "tenant_store_db", "initial_switch"))
+                    .willReturn(initialTenantStoreLiquibase);
+            given(liquibaseFactory.create(defaultTenantDataSource, "tenant_store_db")).willReturn(tenantStoreLiquibase);
+
+            given(tenantDetailsService.findAllTenants()).willReturn(allTenants);
+            given(tenantDataSourceFactory.create(defaultTenant)).willReturn(defaultTenantDataSource);
+            given(liquibaseFactory.create(defaultTenantDataSource, "tenant_db", "initial_switch")).willReturn(initialTenantLiquibase);
+            given(liquibaseFactory.create(defaultTenantDataSource, "tenant_db")).willReturn(tenantLiquibase);
+        });
+        Given("Liquibase runs the very first time for the tenant store", () -> {
+            given(databaseStateVerifier.isFirstLiquibaseMigration(defaultTenantDataSource)).willReturn(true);
+        });
+        Given("A previously Flyway migrated tenant store database", () -> {
+            given(databaseStateVerifier.isFlywayPresent(defaultTenantDataSource)).willReturn(true);
+        });
+        Given("A previously Flyway migrated tenant store database on an earlier version than 1.6", () -> {
+            given(databaseStateVerifier.isTenantStoreOnLatestUpgradableVersion(defaultTenantDataSource)).willReturn(false);
+        });
+        Given("Liquibase runs the very first time for the default tenant", () -> {
+            given(databaseStateVerifier.isFirstLiquibaseMigration(defaultTenantDataSource)).willReturn(true);
+        });
+        Given("A previously Flyway migrated default tenant database", () -> {
+            given(databaseStateVerifier.isFlywayPresent(defaultTenantDataSource)).willReturn(true);
+        });
+        Given("A previously Flyway migrated default tenant database on the 1.6 version", () -> {
+            given(databaseStateVerifier.isTenantStoreOnLatestUpgradableVersion(defaultTenantDataSource)).willReturn(true);
+        });
+
+        When("The database migration process is executed", () -> {
+            // TODO: @galovics not sure if this is the action you are expecting here
+            if (!disabled) {
+                liquibaseFactory.create(defaultTenantDataSource, "tenant_db", "initial_switch");
+            }
+
+            // NOTE: this is how exceptions are tested
+            // exception = assertThrows(LiquibaseException.class, () -> {
+            // });
+        });
+
+        Then("The tenant store upgrade fails with a schema upgrade needed", () -> {
+            // TODO: @galovics not sure what "result" you are referencing here
+            // assertThat(result).isNotNull();
+
+            if (disabled) {
```

Comment:
Again, no differentiation needed. If this `Then` says it should fail, let's verify the failure.

Reply:
Not sure if you saw the parameterized input... technically that would cover for mock Liquibase enabled and disabled...